lib/teleterm/clusters/dbcmd_cli_command_provider.go

```diff
@@ -27,42 +27,33 @@ import (
 // DbcmdCLICommandProvider provides CLI commands for database gateways. It needs Storage to read
 // fresh profile state from the disk.
 type DbcmdCLICommandProvider struct {
-	storage StorageByResourceURI
+	cluster *Cluster
 	execer  dbcmd.Execer
 }
 
-type StorageByResourceURI interface {
-	GetByResourceURI(string) (*Cluster, error)
-}
-
-func NewDbcmdCLICommandProvider(storage StorageByResourceURI, execer dbcmd.Execer) DbcmdCLICommandProvider {
+func NewDbcmdCLICommandProvider(cluster *Cluster, execer dbcmd.Execer) DbcmdCLICommandProvider {
 	return DbcmdCLICommandProvider{
-		storage: storage,
+		cluster: cluster,
 		execer:  execer,
 	}
 }
 
 func (d DbcmdCLICommandProvider) GetCommand(gateway *gateway.Gateway) (*exec.Cmd, error) {
-	cluster, err := d.storage.GetByResourceURI(gateway.TargetURI())
```

Comment:
When connecting to the local proxy, `DbcmdCLICommandProvider` only needs `tc.SiteName` (for things like psql profile name) and maybe `tc.InsecureSkipVerify`. TeleportClient and cluster.status are used more when not connecting to the local proxy. 

One exception to above is Oracle where `profile.OracleWalletDir` is called by `DbcmdCLICommandProvider`. But I don't think Oracle works anyway in Connect since custom logic is needed to create that OracleWalletDir in the first plac.e

So maybe I can refactor `DbcmdCLICommandProvider` so that we just pass the cluster name here. What do guys think?

Reply:
`dbcmd` depending on `TeleportClient` instead of on a more specific data structure feels like an oversight which just naturally happened over time. I think it might be another side effect of tsh not having to care about long-lived `TeleportClient` instances – why make `dbcmd` depend on something else than `TeleportClient` when you can just read stuff from `tc` and you can be sure that it's fresh on each `tsh` call.

This might be another argument for returning a fresh `TeleportClient` in dbcmd command builder, so that the semantics of using `TeleportClient` in Connect resemble those of using `TeleportClient` in tsh. I don't have a horse in this race though so feel free to pick what works best.